core-library/src/main/java/org/silverpeas/core/index/search/model/IndexSearcher.java

```diff
@@ -580,4 +569,18 @@ protected PrefixQuery getTermQueryOnFolder(QueryDescription query) {
     Term term = new Term(IndexManager.PATH, query.getRequestedFolder());
     return new PrefixQuery(term);
   }
+
+  private TermRangeQuery getTermRangeQuery(String fieldName, String beginDate, String endDate) {
+    if (!StringUtil.isDefined(beginDate) && !StringUtil.isDefined(endDate)) {
+      return null;
+    }
+
+    if (!StringUtil.isDefined(beginDate)) {
+      beginDate = IndexEntry.STARTDATE_DEFAULT;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Introduce a new variable instead of reusing the parameter "beginDate". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1226)